coco/request_forwarder.py

```diff
@@ -72,9 +108,9 @@ def add_endpoint(self, name, endpoint):
         """
         return await self._endpoints[name].call(request)
 
-    @staticmethod
-    async def _request(session, method, host, endpoint, request):
+    async def _request(self, session, method, host, endpoint, request):
         url = host + endpoint
+        host_label, port = format_metric_label(host)
```

Comment:
Not sure if we shouldn't move this out of here and only do it once per host.

Reply:
If we change this as described above and then do #24, keeping that in a `Host` object could make sense.

Reply:
Do you mean instead of creating a new `session.request` for every host every time, just store a list of `ClientSession` (or a wrapper around that with the labels) in `self._groups`?

Reply:
That could be a first step, but if then we also store hostname and port seperately, we might want to refactor all 3 of those into a `Host` class.

Reply:
Yeah, that sounds good. It won't be much work so I can put it in now, unless you would prefer it in a separate PR

Reply:
No, that would be wonderful!

Reply:
actually, looking a bit more closely it would probably be best to leave that for a separate PR.

Reply:
OK, well since you sounded enthusiastic just as I backed out I will give it a go! 😅

Reply:
No, don't worry. Whatever makes more sense... Nothing wrong by doing it in another PR